File: backend/src/main/java/com/fitgym/model/FoodSuggestion.java
package com.fitgym.model;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Document(collection = "food_suggestions")
@Data
public class FoodSuggestion {
    @Id
    private String id;
    private String userId;
    private String mealType;
    private String name;
    private String description;
    private Integer calories;
    private Double protein;
    private Double carbs;
    private Double fats;
    private String fitnessGoal;
}

File: backend/src/models/FoodSuggestion.js
const mongoose = require('mongoose');

const foodSuggestionSchema = new mongoose.Schema(
  {
    userId: String,
    mealType: String,
    name: { type: String, required: true, trim: true },
    description: String,
    calories: Number,
    protein: Number,
    carbs: Number,
    fats: Number,
    fitnessGoal: { type: String, default: 'general' }
  },
  {
    collection: 'food_suggestions',
    toJSON: {
      virtuals: true,
      transform: (_doc, ret) => {
        ret.id = ret._id.toString();
        delete ret._id;
        delete ret.__v;
        return ret;
      }
    }
  }
);

module.exports = mongoose.model('FoodSuggestion', foodSuggestionSchema);
